editor/src/document/document_file.rs

```diff
@@ -568,6 +574,45 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					responses.push_back(DocumentOperation::DuplicateLayer { path }.into());
 				}
 			}
+			SelectLayer(selected, ctrl, shift) => {
+				let mut paths = vec![];
+				let last_selection_exists = !self.layer_range_selection_reference.is_empty();
+
+				// If we don't have ctrl selected, clear last selection
+				if !ctrl {
+					// Todo: This clears the selection, figure out how to improve this
+					self.layer_data.iter_mut().filter(|(_, layer_data)| layer_data.selected).for_each(|(path, layer_data)| {
+						layer_data.selected = false;
+						responses.push_back(LayerChanged(path.clone()).into())
+					});
```

Comment:
responses.push_back(DeselectAllLayers.into());

Reply:
I would love this to work, but this is what I tried initially and it simply doesn't propagate the message in time it seems. It happens too late and deselects everything right after you select.

Reply:
have you tried enabling the debug output to trace the messages?

Reply:
Ah that is due to the push_front you did down there

Reply:
Oh, this one? _responses.push_front(AddSelectedLayers(paths).into());_. Let me try switching that around.

Reply:
I was following the structure of 
```
SetSelectedLayers(paths) => {
self.layer_data.iter_mut().filter(|(_, layer_data)| layer_data.selected).for_each(|(path, layer_data)| {
	layer_data.selected = false;
	responses.push_back(LayerChanged(path.clone()).into())
});

responses.push_front(AddSelectedLayers(paths).into());
}
```

Reply:
Hmm, I tried reordering the responses there. Making both push back, and other combinations with push front. The selection is somewhat functional when they both push back, but the shift select doesn't work properly. I'll investigate this more.

Reply:
you can always press iirc `3` to open the message debugging (in the dev console) that should help to figure this out

Reply:
Resolved this issue by leaning on the function, _SetSelectedLayers_, that already has the clear logic, ha! 

```
// Add or set our selected layers
if ctrl {
	responses.push_front(AddSelectedLayers(paths).into());
} else {
	responses.push_front(SetSelectedLayers(paths).into());
}
```

As opposed to clearing before hand.